src/main/java/ai/elimu/web/project/app_category/AppCategoryDeleteController.java

```diff
@@ -35,9 +36,16 @@ public String handleRequest(Model model, @PathVariable Long projectId, @PathVari
         
         AppCategory appCategory = appCategoryDao.read(id);
         model.addAttribute("appCategory", appCategory);
-        appCategoryDao.delete(appCategory);
+        
+        try {
 
-        return "redirect:/project/app-category/list";
+            appCategoryDao.delete(appCategory);
+            return "redirect:/project/app-category/list";
+        } catch (DataAccessException dae) {
+        	// TODO implement error message
+        	// return error message
+        	return "redirect:/project/app-category/list";
```

Comment:
If an error occurs, the user should remain on the same page instead  of being redirected back to the list, right?